src/SFA.DAS.ApprenticeAccounts/Application/Commands/UpdateApprenticePreferenceCommand/UpdateApprenticePreferenceCommandHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.ApprenticeAccounts.Data;
+using SFA.DAS.ApprenticeAccounts.Data.Models;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticeAccounts.Application.Commands.UpdateApprenticePreferenceCommand
+{
+    public class UpdateApprenticePreferenceCommandHandler : IRequestHandler<UpdateApprenticePreferenceCommand>
+    {
+        private readonly IApprenticePreferencesContext _apprenticePreferencesContext;
+        private readonly IApprenticeContext _apprenticeContext;
+        private readonly IPreferencesContext _preferencesContext;
+        private readonly ILogger<UpdateApprenticePreferenceCommandHandler> _logger;
+
+        public UpdateApprenticePreferenceCommandHandler(IApprenticePreferencesContext apprenticePreferencesContext,
+            IApprenticeContext apprenticeContext, IPreferencesContext preferencesContext, ILogger<UpdateApprenticePreferenceCommandHandler> logger)
+        {
+            _apprenticePreferencesContext = apprenticePreferencesContext;
+            _apprenticeContext = apprenticeContext;
+            _preferencesContext = preferencesContext;
+            _logger = logger;
+        }
+
+        public async Task<Unit> Handle(UpdateApprenticePreferenceCommand request, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Fetch Apprentice record by Id: {request.ApprenticeId}");
```

Comment:
You may be able to put this code into a mediator validator as well. As it is a pre-requisite to the action. It essentially runs before the handler by checking the validation first and only runs the handler if it passes. It's akin to this: https://medium.com/swlh/validation-with-mediator-pipelines-in-asp-net-core-applications-7878a56ec604 and you will probably find it across some of our other projects. It might make the handler a bit nicer.